core/domain/rights_manager.py

```diff
@@ -644,6 +648,17 @@ def _publish_activity(committer_id, activity_id, activity_type):
         committer_id, activity_id, activity_type, ACTIVITY_STATUS_PUBLIC,
         '%s published.' % activity_type)
 
+    # Add contribution dates to the owners and editors. This assumes that the
+    # only people who could possibly make commits to the exploration are those
+    # listed explicitly as owners/editors of the exploration.
+    activity_rights = _get_activity_rights(activity_type, activity_id)
+    for owner_id in activity_rights.owner_ids:
+        user_services.update_first_contribution_msec_if_not_set(
+            owner_id, utils.get_current_time_in_millisecs())
+    for editor_id in activity_rights.editor_ids:
```

Comment:
For the ExplorationSnapshotMetadataModel, is there a way to cycle through only the snapshots for a specific exploration? Or do I cycle through the entire model and look for snapshot_ids that corresponding to the specific exploration id?